plugins/module_utils/parsing.py

```diff
@@ -11,7 +11,7 @@
 METRIC_SUFFIXES = ["K", "M", "G", "T", "P"]
 
 duration_pattern = re.compile(
-    r"^((?P<Y>[1-9]\d*)Y)?((?P<W>[1-9]\d*)W)?((?P<D>[1-9]\d*)D)?(((?P<H>[1-9]\d*)H)?((?P<M>[1-9]\d*)M)?)?$"
+    r"^0*((?P<Y>[1-9]\d*)Y)?((?P<W>[1-9]\d*)W)?((?P<D>[1-9]\d*)D)?(((?P<H>[1-9]\d*)H)?((?P<M>[1-9]\d*)M)?)?$"
```

Comment:
This actually allows leading zeroes only in front of the whole ISO8601 and not in the middle, which seems kinda weird to me. It does not match [this for example](https://regex101.com/r/JhnhHX/1). Maybe it would make more sense to just simplify the `[1-9]\d*` group to just `\d+`?

Reply:
Also it would probably make sense from usability standpoint to also allow lowercase units (I think it does only uppercase?) and spaces between the groups, for example "1h 1m". But I mean that only as an observation, it is beyond the scope of this ticket. Leading zeroes are a more direct concern.

Reply:
It will work for lower case units - it first converts lower case to upper case (`duration_pattern.match(period.upper())`)

Reply:
Good point, will fix it now.